src/store-api/src/metadata.rs

```diff
@@ -153,6 +153,12 @@ pub struct RegionMetadata {
 
     /// Primary key encoding mode.
     pub primary_key_encoding: PrimaryKeyEncoding,
+
+    /// Partition expression serialized as a JSON string.
+    /// Compatibility behavior:
+    /// - None: no partition expr was ever set in the manifest (legacy regions).
+    /// - Some(""): an explicit “single-region/no-partition” designation. This is distinct from None and should be preserved as-is.
+    pub partition_expr: Option<String>,
```

Comment:
Is it possible to keep the deserialized expr in the metadata?

Reply:
I hope so, but can't due to the dep cycle.